app/celery/service_callback_tasks.py

```diff
@@ -121,6 +127,8 @@ def create_delivery_status_callback_data(notification, service_callback_api):
         "notification_type": notification.notification_type,
         "service_callback_api_url": service_callback_api.url,
         "service_callback_api_bearer_token": service_callback_api.bearer_token,
+        "template_id": str(notification.template_id),
```

Comment:
The `create_delivery_status_callback_data` function is used to encrypt the data before it gets used as an argument to the task (because it contains email address or phone number). Without encrypting it, it's possible to see those values in SQS because the message is only base64 encoded.

Reply:
Would it be equivalent to just pass the `notification_id` to the task? It looks like we can get all the data we need from that, including the service callback.

I get that we need to encrypt stuff on SQS, but it looks like we're doing more work than we need to - we wouldn't need to encrypt the notification ID if we just sent that as the data for the task.

To be clear, I'm not suggesting we change this here, but it would be helpful to simplify it as it made the PR harder to understand - we're making the change twice.

Reply:
I see what you mean about having two dicts and how it would be nice to simplify things and avoid that. If we were just to pass the notification ID to the task it would make the code simpler, but the big downside would be that the task would then need to make another database query to get the `status`, `client_reference` etc.

Reply:
That's true. I assume by big downside you're referring to the fact that we process _a lot_ of callbacks, and that this would effectively be an extra DB query, since the parent tasks still need to retrieve the notification for other purposes. Have I understood that right? If so, I think I understand the purpose of this task better now: to protect the parent task from the remote API call.

Even then, it's still seems unnecessary to have two different styles of hash, but also not a big issue. I'm pleased I've got a bit more familiar with the subtleties of this bit of the codebase.

(Feel free to merge 👍)

Reply:
Yes, I was referring to the extra database query per callback that we'd need if we only passed through the notification_id. I'll deploy this as is for now, and we can always come back to maybe trying to make the dictionaries the same another time.